lua/nvim-navic/lib.lua

```diff
@@ -254,24 +254,15 @@ function M.request_symbol(for_buf, handler, client_id, file_uri)
 		}
 	end
 
-	vim.lsp.buf_request_all(
-		for_buf,
-		"textDocument/documentSymbol",
-		{ textDocument = textDocument_argument },
-		function(symbols)
-			if symbols[client_id] == nil then
-				return
-			elseif symbols[client_id].error ~= nil or symbols[client_id].result == nil then
-				vim.defer_fn(function()
-					M.request_symbol(for_buf, handler, client_id)
-				end, 750)
-			elseif symbols[client_id].result ~= nil then
-				if vim.api.nvim_buf_is_valid(for_buf) then
-					handler(for_buf, symbols[client_id].result)
-				end
+	client.request("textDocument/documentSymbol", { textDocument = textDocument_argument }, function(err, symbols, _, _)
+		if err ~= nil then
+			return
+		elseif symbols ~= nil then
+			if vim.api.nvim_buf_is_valid(for_buf) then
+				handler(for_buf, symbols)
```

Comment:
Can the return be passed directly or does it have `result` field?

Reply:
I'm not sure I understand what you mean with this? 😅 

The result that comes from lsp does not have a `result` field since that is a separate argument in this case. The callback have the signature of `(err, result, ctx)`

https://github.com/neovim/neovim/blob/75e1b1d5c934aa3a12693a7df58795563fb7877e/runtime/lua/vim/lsp.lua#L1399